### DISCUSSION TITLE
Upgrading to 4.9.3

### DISCUSSION THREAD

**Discussion by renzosilv** [antlr/antlr4#3552] [Q&A]
I'm fairly new to antlr and I inherited a project at work which uses it - my question is two fold.  Currently in nuget I only see Antlr4.Runtime.Standard but I have currently installed Antlr4, Antlr4.Runtime and Antlr4.Codegenerator (4.6.6) 
Do I now only need the standard package?  Is there a place where I can learn how to make the upgrade?

The reason I want to upgrade is because currently I'm core dumping with files  larger than 1 gig (object is 4 gigs in memory)  and running in an environment with plenty more.   I gather from reading some of the issues that it might help since it uses a newer version of .net?  

Thanks in advance.

**Comment by ericvergnaud** [comment]
It seems you are using the visiontunnellabs fork, not the official antlr release.

**Reply by kaby76** [reply]
Although it is not "official", you can replace Antl4/Antlr4.Runtime with my [Antlr4BuildTasks](https://www.nuget.org/packages/Antlr4BuildTasks/) plus the "official" [Antlr4.Runtime.Standard](https://www.nuget.org/packages/Antlr4.Runtime.Standard/) Nuget libraries. Antlr4BuildTasks is a replacement for [antlr4](https://www.nuget.org/packages/Antlr4/)--I rewrote part of that fork. It is a wrapper for the "official" Java Antlr4 4.9.3 tool jar, and comes with a stand-alone JRE, and a copy of the Antlr4.9.3 complete jar, all within the Nuget package. The MSBuild/Dotnet rules to build the parser from the grammar are almost the same as the Antlr4 Nuget package. You don't need Java; You don't need to download Antlr4 complete jar. The replacement for [Antlr4.Runtime](https://www.nuget.org/packages/Antlr4.Runtime/) is the [Antlr4.Runtime.Standard](https://www.nuget.org/packages/Antlr4.Runtime.Standard/) package. If you use Antlr4BuildTasks and Antlr4.Runtime.Standard, do not reference Antlr4 nor Antlr4.Runtime. The Tunnelvision fork is very old. There is no replacement for [Antlr4.CodeGenerator](https://www.nuget.org/packages/Antlr4.CodeGenerator/), so just remove that from your .csproj file. The instructions for Antlr4BuildTasks is [here](https://github.com/kaby76/Antlr4BuildTasks).
